admin/tests/integration/components/PasswordComponent.spec.js

```diff
@@ -0,0 +1,123 @@
+import { mount } from "@vue/test-utils";
+import { beforeEach, describe, expect, it } from "vitest";
+
+import PasswordComponent from "@/components/PasswordComponent.vue";
+
+describe("Integration | Components | PasswordComponent", () => {
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = mount(PasswordComponent);
+  });
+
+  it("should display default label", () => {
+    // then
+    const label = wrapper.find("label");
+    expect(label.text()).toBe("mot de passe");
+  });
+
+  it("should display custom label", () => {
+    // given
+    wrapper = mount(PasswordComponent, {
+      props: {
+        label: "New Password",
+      },
+    });
+
+    // then
+    const label = wrapper.find("label");
+    expect(label.text()).toBe("New Password");
+  });
+
+  it("should have password input type by default", () => {
+    // then
+    const input = wrapper.find("input");
+    expect(input.attributes("type")).toBe("password");
+  });
+
+  it("should toggle password visibility when button is clicked", async () => {
+    // given
+    const input = wrapper.find("input");
+    const button = wrapper.find("button");
+
+    // when
+    await button.trigger("click");
+
+    // then
+    expect(input.attributes("type")).toBe("text");
+  });
+
+  it("should toggle back to password type when button is clicked twice", async () => {
+    // given
+    const input = wrapper.find("input");
+    const button = wrapper.find("button");
+
+    // when
+    await button.trigger("click");
+
+    // then
+    expect(input.attributes("type")).toBe("text");
+
```

Comment:
You can directly update variable of showPassword with this line
```suggestion
    wrapper.vm.showPassword = true;

```